### ISSUE TITLE
Isomers in the databank

### ISSUE THREAD

**Issue by ohsOllila** [NMRLipids/NMRlipidsIVPEandPG#41]
Different enantiomers and separating pro-S and pro-H hydroges have been discussed many times during the NMRlipids project, most recently in issues https://github.com/NMRLipids/NMRlipidsIVPEandPG/issues/38 and https://github.com/patrickfuchs/buildH/issues/63. 

I think that we have to soon decide if the mapping files used in the databank will contain enantiometric information or not.

**Comment by patrickfuchs** [comment]
Quoting your remark in issue #38 :

> Idea is that we would not use the def files anymore in the databank. All the information should be read from the mapping files. We can continue the discussion in the new issue, but I think that we also need a online meeting about the databank soon.

Indeed that would be nice to discuss this so that buildH uses the same input files as NMRlipids. So far, buildH uses the `.def` files to infer what C-H the user wants, to get the atom (PDB) name of those C-H, and last to get their generic name (such as alpha1, alpha2, etc., for the output). If you plan to stop using `.def` files, it'll have consequences for buildH, so it's important that we talk about it soon.

**Comment by ohsOllila** [comment]
Maybe @akiirik could comment how this is implemented in practice now?

One solution may be that the def files are created from mapping files if needed. Therefore, using def files in buildH is not necessarily a problem if you want to continue using them.

Because mapping files include the same information as the def files, I do not want to ask NMRlipids databank contributors to deliver both mapping file and def file. Due to overlapping information, getting rid of def files completely is also an option in the future.

**Comment by patrickfuchs** [comment]
> One solution may be that the def files are created from mapping files if needed. Therefore, using def files in buildH is not necessarily a problem if you want to continue using them.

Not especially. We decided using them because we thought it was still used in NMRlipids. The only extra information we need next to mapping files is the generic names, but this can be easily added (and only once per lipid, so this is less work). The only thing I'm wondering is: so far the `.def`files  are used to guess what C-H the user wants to calculate the OP on, for example if he/she provides only the polar head, buildH only compute things on the polar head. This behavior is inherited from @jmelcr script. Now, I have to see how to implement this if we use mapping files.

> Because mapping files include the same information as the def files, I do not want to ask NMRlipids databank contributors to deliver both mapping file and def file. Due to overlapping information, getting rid of def files completely is also an option in the future.

Sure I understand. Since mapping file are more general, I think that would be a good option also for buildH.  We'll think about it.

**Comment by ohsOllila** [comment]
> The only thing I'm wondering is: so far the .deffiles are used to guess what C-H the user wants to calculate the OP on, for example if he/she provides only the polar head, buildH only compute things on the polar head.

In the databank, all order parameters are calculated automatically so user do not make such decisions. However, I am not sure how this is implemented in practice now. Maybe @akiirik can answer to this.

**Comment by akiirik** [comment]
The definition file needed by buildH is generated from the mapping file. Every C-H bond is put in the definition file, so there is no option to choose a specific part of the molecule.

**Comment by patrickfuchs** [comment]
OK, thanks for the comment @akiirik.

**Comment by ohsOllila** [comment]
We can continue this discussion in the issue that I have created into the Databank repository: https://github.com/NMRLipids/Databank/issues/1. Therefore, I will close this issue.
